source/isaaclab/isaaclab/devices/spacemouse/se2_spacemouse.py

```diff
@@ -3,152 +3,117 @@
 #
 # SPDX-License-Identifier: BSD-3-Clause
 
-"""Spacemouse controller for SE(2) control."""
-
-import hid
-import numpy as np
-import threading
-import time
-from collections.abc import Callable
-
-from ..device_base import DeviceBase
-from .utils import convert_buffer
-
-
-class Se2SpaceMouse(DeviceBase):
-    r"""A space-mouse controller for sending SE(2) commands as delta poses.
-
-    This class implements a space-mouse controller to provide commands to mobile base.
-    It uses the `HID-API`_ which interfaces with USD and Bluetooth HID-class devices across multiple platforms.
+# Copyright (c) 2022, NVIDIA CORPORATION & AFFILIATES, ETH Zurich, and University of Toronto
+# Copyright (c) 2023, Boston Dynamics AI Institute, Inc.
+# All rights reserved.
+#
+# SPDX-License-Identifier: BSD-3-Clause
 
-    The command comprises of the base linear and angular velocity: :math:`(v_x, v_y, \omega_z)`.
+"""SpaceMouse controller for SE(2) control."""
 
-    Note:
-        The interface finds and uses the first supported device connected to the computer.
+import numpy as np
 
-    Currently tested for following devices:
+from .base_spacemouse import SpaceMouseBase
 
-    - SpaceMouse Compact: https://3dconnexion.com/de/product/spacemouse-compact/
 
-    .. _HID-API: https://github.com/libusb/hidapi
+class Se2SpaceMouse(SpaceMouseBase):
+    """A SpaceMouse controller for sending SE(2) commands as delta poses.
+    The use-case of this class is to control the robot in SE(2) space (x, y, yaw),
+    where theta is the rotation about the z-axis.
+    This is useful for controlling a differential drive robot.
 
     """
 
-    def __init__(self, v_x_sensitivity: float = 0.8, v_y_sensitivity: float = 0.4, omega_z_sensitivity: float = 1.0):
-        """Initialize the spacemouse layer.
+    def __init__(
+        self,
+        v_x_sensitivity: float = 0.8,
+        v_y_sensitivity: float = 0.4,
+        omega_z_sensitivity: float = 1.0,
+        interval_ms: float = 10.0,
+    ):
+        """Initialize the SpaceMouse layer.
 
         Args:
-            v_x_sensitivity: Magnitude of linear velocity along x-direction scaling. Defaults to 0.8.
-            v_y_sensitivity: Magnitude of linear velocity along y-direction scaling. Defaults to 0.4.
-            omega_z_sensitivity: Magnitude of angular velocity along z-direction scaling. Defaults to 1.0.
+            v_x_sensitivity (float): Magnitude of linear velocity along x-direction scaling. Defaults to 0.8.
+            v_y_sensitivity (float): Magnitude of linear velocity along y-direction scaling. Defaults to 0.4.
+            omega_z_sensitivity (float): Magnitude of angular velocity along z-direction scaling. Defaults to 1.0.
+            interval_ms (float): Update interval for the SpaceMouse in milliseconds. Defaults to 10ms (100Hz).
```

Comment:
Need to remove type annotations in docstrings everywhere.

```suggestion
            v_x_sensitivity: Magnitude of linear velocity along x-direction scaling. Defaults to 0.8.
            v_y_sensitivity: Magnitude of linear velocity along y-direction scaling. Defaults to 0.4.
            omega_z_sensitivity: Magnitude of angular velocity along z-direction scaling. Defaults to 1.0.
            interval_ms: Update interval for the SpaceMouse in milliseconds. Defaults to 10ms (100Hz).
```